### PR TITLE
feature: validate swap input against current liquidity protection

### PR DESCRIPTION
**Pull request by alanrsoares**
image

### CODE REVIEW

**Review by oieduardorabelo — APPROVED** [review]
lgtm

File: app/src/business/calculators/swapCalculatorPMTP.ts [hunk 1]
```diff
@@ -394,12 +407,28 @@ export function useSwapCalculator(input: {
       return SwapState.INSUFFICIENT_LIQUIDITY;
     }
 
-    // slippage > 50% can be social engineered as a non-option as to prevent traders from transacting without understanding the potential price volatility before their transaction will actually execute
-    // user entering a negative slippage is not useful but hopefully works out for them
+    if (
+      fromField.value.asset?.symbol === "rowan" &&
+      fromField.value.fieldAmount.greaterThanOrEqual(
```
Comment (canercandan) [inline app/src/business/calculators/swapCalculatorPMTP.ts:412]: @alanrsoares it should actually go though if the sold amount of rowan is equal to `currentRowanLiquidityThreshold`

also a side note `currentRowanLiquidityThreshold` includes the 18 decimals something to be aware of when comparing.

Comment (alanrsoares) [reply app/src/business/calculators/swapCalculatorPMTP.ts:412]: the decimals are being converted here @canercandan. Here I'm checking if the value the user is trying to swap from (selling) is `>=` the current threshold

Comment (canercandan) [reply app/src/business/calculators/swapCalculatorPMTP.ts:412]: @alanrsoares yep but LIQP will actually work if the amount of rowan sold is less or equal to the threshold.

**Comment by bksifchain** [comment]
@alanrsoares 

I know its not our current practice on v2 (since we’ve been putting helper text on the buttons) … but my suggestion would be to only display the warning/error copy above the button. 

Warning message
- Use red border/text - since you are blocked / its an error.
- Copy: The current max swap amount is 83 ROWAN. This value updates every block (~6 seconds).

CTA
- Copy: `Swap`
- State: Disabled

**Review by oieduardorabelo — APPROVED** [review]

**Pull request merged**